pkg/osv/osv.go

```diff
@@ -306,29 +291,54 @@ func HydrateWithClient(resp *BatchedResponse, client *http.Client) (*HydratedBat
 	return &hydrated, nil
 }
 
-// makeRetryRequest will return an error on both network errors, and if the response is not 200
+// makeRetryRequest executes HTTP requests with exponential backoff retry logic
 func makeRetryRequest(action func() (*http.Response, error)) (*http.Response, error) {
-	var resp *http.Response
-	var err error
+	const maxRetryAttempts = 4
```

Comment:
These two variables already exist at the top const section

Reply:
yeah, I kept changing the variable so i wanted to use it locally. I will revert these changes as well